image-pipeline/expression-scatter-plots.js

```diff
@@ -7,30 +7,42 @@
  *
  * node expression-scatter-plots.js --accession="SCP24" # Staging, 1.3M cell study
  */
-import { parseArgs } from 'node:util'
 import { mkdir, writeFile, readFile, access } from 'node:fs/promises'
-import sharp from 'sharp'
 import os from 'node:os'
+import { exit } from 'node:process'
+import { parseArgs } from 'node:util'
 
+import { gunzipSync, strFromU8 } from 'fflate'
 import puppeteer from 'puppeteer'
-import { exit } from 'node:process'
+import sharp from 'sharp'
 
 const args = process.argv.slice(2)
 
 const options = {
-  accession: { type: 'string' }
+  'accession': { type: 'string' }, // SCP accession
+  'cluster': { type: 'string' }, // Name of clustering
+  'cores': { type: 'string' }, // Number of CPU cores to use. Default: all - 1
+  'debug': { type: 'boolean' }, // Whether to show browser UI and DevTools
+  'environment': { type: 'string' }, // development, staging, or production
+  'json-dir': { type: 'string' } // Path to expression arrays; for development
 }
 const { values } = parseArgs({ args, options })
 
+const timeoutMinutes = 10 // 0.75
+
 // Candidates for CLI argument
 // CPU count on Intel i7 is 1/2 of reported, due to hyperthreading
-// const numCPUs = os.cpus().length / 2 - 1
-const numCPUs = 1 // For easier local development
+const numCPUs = values.cores ? parseInt(values.cores) : os.cpus().length / 2 - 1
 console.log(`Number of CPUs to be used on this client: ${numCPUs}`)
 
 // TODO (SCP-4564): Document how to adjust network rules to use staging
-// const origin = 'https://singlecell-staging.broadinstitute.org'
-const origin = 'https://localhost:3000'
+const originsByEnvironment = {
+  'development': 'https://localhost:3000',
+  'staging': 'https://singlecell-staging.broadinstitute.org',
+  'production': 'https://singlecell.broadinstitute.org'
+}
+const environment = values.environment ?? 'development'
```

Comment:
`??` only looks at explicitly `null` or `undefined` so an empty string would get through in this case. Is that a reasonable value for `environment` to be set to here, or would it better to check for any falsy and do `const environment = values.environment || 'development'` ?

Reply:
Nice catch!  Done in 3472483.